### PR TITLE
Simplification refactoring

### PR DESCRIPTION
**Pull request by GeorgSchneider**
This PR largely simplifies the template. It's a single commit, so better to be checked out for review - apologies. The following changes are in there:
- Removed JS version of the UI: in the long run it's too much effort to maintain both versions at parity and we should be pushing people to use the TS version as it's much less error-prone. They could still write components in JS if they wish to do so
- Removed `Contracts` component: this component has grown to unmanageable complexity, and it suggests to be used off-the-shelf, which it shouldn't.
- Remove Default page as it didn't really show anything useful
- Simplified Header, Sidebar, and Layout component
- The `Report` page now shows an explicit table with choices enabled depending on the party. This is the example that's meant to be copied and modified. Having the table explicit in there allows people to modify this to their needs, almost as simple as the previous, complex generic `Contracts` component.
- The multi-parameter dialog for choice parameters has been factored out into its own component. It supports text, number, date, and selection inputs. Number, date and selection are shown in the example
- DAML model has been extended with a choice to demonstrate multi-parameter exercise

### CODE REVIEW

File: ui/src/pages/report/InputDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+export interface InputDialogProps {
+  open : boolean
+  title : string
+  fields : Field[]
+  onClose : (state : any) => Promise<void>
+}
+
+export const InputDialog = (props : InputDialogProps) => {
+  const [ state, setState ] = useState<any>({});
```
Comment (leonidr-da) [inline ui/src/pages/report/InputDialog.tsx:31]: I wish that we didn't need this `any` and use a type parameter for the type of the choice. There was a bug in the original Dialog code that could have been avoided if we did that. Fine the way that it is.

File: ui/src/pages/report/InputDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  return (
+    <Dialog open={props.open} onClose={() => props.onClose(undefined)} maxWidth="sm" fullWidth>
+      <DialogTitle>
+        {props.title}
+      </DialogTitle>
+      <DialogContent>
+        {props.fields.map((f, i) => {
+          if ((f.type as Selection).items) {
```
Comment (leonidr-da) [inline ui/src/pages/report/InputDialog.tsx:40]: Oh cool, I didn't realize that you can do that! That's an elegant way to check, but is a shame that the Typescript compiler can't reuse that 4 lines below.

File: ui/src/pages/report/Report.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  const ledger = useLedger();
+  const assets = useStreamQuery(Asset).contracts;
+  
+  const showGive = (asset : CreateEvent<Asset>) => {
+    const fields : Field[] = [
+      { label: "New Owner", name: "newOwner", type: { items: [ "Alice", "Bob" ] } }
+    ]
+    const onClose = async (state : any) => {
+      console.log(state);
+      setProps({ open: false, title: "", fields: [], onClose: async () => {} });
```
Comment (leonidr-da) [inline ui/src/pages/report/Report.tsx:28]: I wish that we didn't have to repeat ourselves here. Also we could have a state for Give and Appraise?

**Comment by leonidr-da** [comment]
@gaborh-da What do you think?

File: ui/package.json [hunk 4]
```diff
@@ -51,5 +42,14 @@
...
   },
-  "proxy": "http://localhost:7575"
+  "proxy": "http://localhost:7575",
+  "devDependencies": {
+    "@types/jsonwebtoken": "^8.5.0",
+    "@types/react": "^16.9.36",
+    "@types/react-dom": "^16.9.8",
+    "@types/react-router-dom": "^5.1.5",
+    "@types/tinycolor2": "^1.4.2",
+    "react-scripts": "^3.4.1",
```
[the remark this answers is not in the document]
Comment (leonidr-da) [reply ui/package.json:52]: Looks like it does, on DABL at least.

File: README.md [hunk 5]
```diff
@@ -9,87 +9,59 @@ This repository contains a simple UI template for a DAML application. It provide
...
-    cd ui-ts
-    yarn install
-    yarn start
+2. Start the sandbox ledger
+```
+daml start
+```
 
-If you change your DAML models and regenerate the JavaScript code, you need to run this in your UI directory:
+3. Start the UI
```
[the remark this answers is not in the document]
Comment (GeorgSchneider) [reply README.md:24]: Good point, I like it better for clarity

File: README.md [hunk 6]
```diff
@@ -9,87 +9,59 @@ This repository contains a simple UI template for a DAML application. It provide
...
-
-    cd ui-js
-    yarn install
-    yarn start
-
-For Typescript:
+1. Build the DAML models and the UI project
+```
+make build
+```
```
[the remark this answers is not in the document]
Comment (GeorgSchneider) [reply README.md:17]: Yeah, and I normally run just `make` locally. But I thought it's more explicit, and if people would add an additional target as the first one, this wouldn't work anymore. Better to be safe here.

File: README.md [hunk 7]
```diff
@@ -9,87 +9,59 @@ This repository contains a simple UI template for a DAML application. It provide
...
-```bash
-daml build -o daml-ui-template.dar
-```
+## Deploying to production
 
-at the root of your repository. Afterwards, open to the DABL website, select the ledger you want to deploy to, go to the "DAML" selection and upload the DAR `daml-ui-template.dar` you have just created.
+Deploying `daml-ui-template` to the hosted DAML platform [project:DABL](http://projectdabl.com/) is straight forward:
 
-To upload the UI, create a ZIP file containing all your UI assets. First build all projects:
+1. Build the deployment artefacts:
```
[the remark this answers is not in the document]
Comment (GeorgSchneider) [reply README.md:53]: I actually looked this up on the DABL website and they had the British spelling iirc :) but agree, should be American

File: ui/src/pages/report/InputDialog.tsx (hunk 1 above)
Comment (GeorgSchneider) [reply ui/src/pages/report/InputDialog.tsx:31]: yeah, but I saw you actually improved this in another PR. will look at that shortly

**Comment by GeorgSchneider** [comment]
Gabor confimed he's good with the changes, so I'll go ahead with the merge. @leonidr-da can you rebase your other PR off master then, and we can decide separately whether to merge that in as well.

**Pull request merged**